app/controllers/public_health_controller.rb

```diff
@@ -315,6 +316,42 @@ def advanced_filter(patients, filters)
           compare_date_end = Chronic.parse(filter[:value][:end])
           patients = patients.where('Date(latest_assessment_at) > ?', compare_date_start).where('Date(latest_assessment_at) < ?', compare_date_end)
         end
+      when 'latest-report-relative'
```

Comment:
This is a _lot_ we need to add to this function whenever we want to support a relative date range for any date we add to the advanced search. I think we could pull this out into a helper for relative date queries. 

To give some ideas, you can sub in the attribute name like so:
 ```patients = patients.where("Date(#{attribute}") = ?', value)```
So you can pull these out into at least 1 helper method - but there's probably more consolidation you could do.

Reply:
done

Reply:
fyi this now causes brakeman errors: https://github.com/SaraAlert/SaraAlert/pull/511/checks?check_run_id=1398211047

Reply:
Yeah I wrote that sort of quickly when just thinking as I typed, but that code as is would be very vulnerable to SQL injection attacks. 

Parameterizing would help as that does some basic formatting that protects against this sort of thing:
```
Patient.where("Date(:attribute) = :value", {attribute: attribute, value: Date.today})
```
but that may not be enough. @holmesie Are you aware of a better way to handle this sort of case so that it's safe?
If there's not an equally safe way to do this, then we may just need to deal with at least some of the duplication.

Reply:
would love to hear @holmesie's thoughts on this.  there were rubocop errors before this change complaining for too much nesting, so complaints either way

Reply:
Have you tried @kylietmo suggestion above? I'm fine with duplication for now, if we have to. You can tell rubocop to ignore that section for that particular complaint.

Reply:
I did, however now I am getting warnings for SQL injection which I think is definitely worse.  I'll revert it to what I had and do 4 helper methods that are basically the same